Feat/document scanner

therealabdi2 opened the pull request:
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] The changes have been tested successfully.
- [x] A changeset has been created (`npm run changeset`).
- [x] I have read and followed the [pull request guidelines](https://capawesome.io/contributing/pull-requests/).

closes https://github.com/capawesome-team/capacitor-mlkit/issues/277

therealabdi2 committed:
feat(document-scanner): implement document scanning functionality

therealabdi2 committed:
feat(document-scanner): add initial release

robingenz commented on packages/document-scanner/android/build.gradle:
```diff
@@ -52,6 +52,7 @@ dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation project(':capacitor-android')
     implementation "androidx.appcompat:appcompat:$androidxAppCompatVersion"
+    implementation 'com.google.android.gms:play-services-mlkit-document-scanner:16.0.0-beta1'
```
Please use a variable for the version like in the other plugins.

robingenz commented on packages/document-scanner/README.md:
````diff
@@ -9,39 +9,127 @@ npm install @capacitor-mlkit/document-scanner
 npx cap sync
 ```
 
+## Supported Platforms
+
+- [x] Android
+- [ ] iOS (Not available)
+- [ ] Web (Not available - native feature)
````
Remove this section and add comments to the methods like in the other plugins.

robingenz commented on packages/document-scanner/README.md:
```diff
@@ -9,39 +9,127 @@ npm install @capacitor-mlkit/document-scanner
+
+## Important Notes
+
+- The ML Kit Document Scanner models, scanning logic, and UI flow are dynamically downloaded by Google Play services. Users might have to wait for these to download before the first use.
```
I guess it would be a good idea to add helepr functions for this like in the Barcode Scanning plugin:
- https://capawesome.io/plugins/mlkit/barcode-scanning/#isgooglebarcodescannermoduleavailable
- https://capawesome.io/plugins/mlkit/barcode-scanning/#installgooglebarcodescannermodule

robingenz commented on packages/document-scanner/package.json:
```diff
@@ -72,7 +72,7 @@
   },
   "peerDependencies": {
-    "@capacitor/core": ">=7.0.0"
+    "@capacitor/core": "^5.0.0 || ^6.0.0 || ^7.0.0"
```
Please revert this change. We only support the latest major version.

robingenz commented on packages/document-scanner/src/web.ts:
```diff
@@ -1,13 +1,18 @@
   extends WebPlugin
   implements DocumentScannerPlugin
 {
   async echo(options: { value: string }): Promise<{ value: string }> {
```
Remove this method.

therealabdi2 committed:
revert(package): pin @capacitor dependencies to version 7.0.0

therealabdi2 committed:
fix(build.gradle): use variable for ML Kit Document Scanner version

therealabdi2 committed:
docs(README): remove unsupported platforms section and clarify Android availability
docs(definitions): add @since annotations for versioning clarity

therealabdi2 committed:
refactor(DocumentScannerWeb): remove unused echo method

therealabdi2 committed:
feat(DocumentScanner): add methods to check and install Google Document Scanner module

therealabdi2 committed:
Fix: Ensure accurate progress reporting for document scanner module installation

- the progress calculation was using integer division, causing intermediate progress values to be reported as 0. This commit corrects the calculation to use floating-point division, ensuring that progress is reported accurately.

robingenz commented:
@therealabdi2 The build fails.

therealabdi2 commented:
> @therealabdi2 The build fails.

@robingenz are you sure the build failing is related to this PR, as I haven't really changed any boilerplate code and its related to ios 

![image](https://github.com/user-attachments/assets/2dd13c13-8520-4cd1-8a0b-a3b814c5d14f)


Could you confirm if the build works on any other PR?

therealabdi2 committed:
fix(Podfile): update paths for Capacitor and CapacitorCordova pods

therealabdi2 commented:
@robingenz try running the build now, i've update the paths in podfile to be consistent with the other packages.
Should probably update the boilerplate template as well

therealabdi2 committed:
fix(settings.gradle): update project directory path for capacitor-android

robingenz commented:
@therealabdi2 Thank you! I will test it myself in the next few days and merge it.

robingenz commented:
@therealabdi2 Are you familiar with Angular? Can you please add a small demo page to the following demo app for this plugin? https://github.com/robingenz/capacitor-mlkit-plugin-demo

therealabdi2 commented:
@robingenz have a look
https://github.com/robingenz/capacitor-mlkit-plugin-demo/pull/33

robingenz reviewed (approved):
LGTM
